src/main/resources/pl/edu/agh/iet/dts/ui/static/js/controller/mapController.js

```diff
@@ -1,11 +1,108 @@
+app.controller('mapController', ['$http', '$scope', '$rootScope', '$interval', function ($http, $scope, $rootScope, $interval) {
+
+    $rootScope.promise = null;
+    $rootScope.lastSync = null;
+
+    $rootScope.period = 60;
+    $rootScope.points = 4;
+    $rootScope.aggregationTime = 10;
+
+    let map = new google.maps.Map(document.getElementById('map'), {
+        center: {lat: 50.04, lng: 19.57},
+        scrollwheel: true,
+        zoom: 8
+    });
+
+    let travelMode = google.maps.TravelMode.WALKING;
+
+    let directionsService = new google.maps.DirectionsService;
+    let directionsDisplay = new google.maps.DirectionsRenderer({
+        map: map
+    });
+
+    let checkpointLimit = 0;
+    let checkpointArray = [];
     
-    $scope.signOut = function () {
-        $location.path('/app/components/login').replace();
+    $rootScope.locationRequest = function () {
+        $http.get('/users/' + $scope.userID + '/positions').then(
+            function (response) {
+                $scope.assignDate();
+                $scope.createRoute(response.data.data);
+            },
+            function (error) {
+                console.log(error);
+            }
+        );
     };
 
-    $scope.showPreferences = function () {
-        $location.path('/app/components/user-form').replace();
+    $scope.assignDate = function () {
+        const format = function (num) {
+            return ('0' + num).slice(-2);
+        };
+
+        now = new Date();        
+        date = now.getFullYear() + '-' + format(now.getMonth() + 1) + '-' + format(now.getDate());
+        time = format(now.getHours()) + ':' + format(now.getMinutes());
+        $rootScope.lastSync = date + ' ' + time;
     };
 
-});
+    $scope.createRoute = function (data) {
+        if (checkpointArray.length === data.length && checkpointLimit === $rootScope.points) {
+            return;
+        }
+
+        checkpointLimit = $rootScope.points;
+        checkpointArray = data.slice();
+        let routeRequest;
+
+        num = Math.min($rootScope.points, 10)
+        data = data.slice(Math.max(0, data.length - num));
+
+        if (data.length === 0) {
+            console.log('Nothing to show. There is no GPS data for this user.');
+        } else if (data.length === 1) {
+            let singlePointCoordinates = new google.maps.LatLng(data[0].latitude, data[0].longitude);
+
+            routeRequest = {
+                origin: singlePointCoordinates,
+                destination: singlePointCoordinates,
+                waypoints: [],
+                travelMode: travelMode
+            };
+        } else {
+            let startPoint = data.shift();
+            let endPoint = data.pop();
+            let wayPoints = [];
+
+            data.forEach(function (wayPoint) {
+                wayPoints.push({
+                    location: new google.maps.LatLng(wayPoint.latitude, wayPoint.longitude),
+                    stopover: wayPoint.stop
```

Comment:
There is no `wayPoint.stop` attribute, therefore `stopover` is always `false`, while it should be definitely `true`. According to the [documentation](https://developers.google.com/maps/documentation/javascript/directions#Waypoints), it is the only way to force the GMaps API to calculate a route through all the waypoints specified, which is surely our case.